conans/client/installer.py

```diff
@@ -289,6 +289,7 @@ def _build_package(self, export_folder, src_folder, build_folder, conan_file, ou
             output.info('Copying sources to build folder')
             shutil.copytree(src_folder, build_folder, symlinks=True)
         os.chdir(build_folder)
+        conan_file._conanfile_directory = build_folder
```

Comment:
Yeah... So...  This was part of another PR, but I think I must have forgot to push this file for whatever reason.  This is to get conan test to work under certain cases where the implementation of the ConanFile is using the conanfile_directory property.